hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -103,14 +103,14 @@ JavaRDD<GenericRecord> toRDD(OffsetRange[] offsetRanges) {
       //Don't want kafka offsets here so we use originalSchemaProvider
       AvroConvertor convertor = new AvroConvertor(originalSchemaProvider.getSourceSchema());
       kafkaRDD = KafkaUtils.<String, byte[]>createRDD(sparkContext, offsetGen.getKafkaParams(), offsetRanges,
-          LocationStrategies.PreferConsistent()).map(obj ->
+          LocationStrategies.PreferConsistent()).filter(obj -> obj.value() != null).map(obj ->
```

Comment:
is this filter redundant, given its getting filtering in 113?

Reply:
nope. here also, we have avro converter from binary. So, we might have to filter it once here. 
and if its not binary, we need to filter null values and thats why.

Reply:
ok